File: WebDriverTests_Java/WebDriverTests/src/test/java/core/supportClass/BySelf.java
package core.supportClass;

import core.systemControls.HtmlControl;
import org.openqa.selenium.By;
import org.openqa.selenium.SearchContext;
import org.openqa.selenium.WebElement;

import java.util.ArrayList;
import java.util.Arrays;

/**
 * Created by Виктория on 23.11.2015.
 */
public class BySelf extends By
{
    private HtmlControl control;

    public BySelf(HtmlControl control)
    {
        this.control = control;
    }

    @Override
    public WebElement findElement(SearchContext context)
    {
        return control.element;
    }

    @Override
    public ArrayList<WebElement> findElements(SearchContext context)
    {
        ArrayList<WebElement> list = control.element != null ? new ArrayList<WebElement>(Arrays.asList(new WebElement[]{control.element})) : new ArrayList<WebElement>();
        return new ArrayList(list);
    }


    @Override
    public String toString()
    {
        return "BySelf";
    }
}

File: WebDriverTests_Csharp/Autotest.Core/Autotests.Utilities/WebTestCore/TestSystem/BySelf.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;
using Autotests.Utilities.WebTestCore.SystemControls;
using OpenQA.Selenium;

namespace Autotests.Utilities.WebTestCore.TestSystem
{
    public class BySelf : By
    {
        private readonly HtmlControl control;

        public BySelf(HtmlControl control)
        {
            this.control = control;
        }

        public override IWebElement FindElement(ISearchContext context)
        {
            return control.element;
        }

        public override ReadOnlyCollection<IWebElement> FindElements(ISearchContext context)
        {
            List<IWebElement> list = control.element != null
                ? new List<IWebElement> {control.element}
                : new List<IWebElement>();
            return new ReadOnlyCollection<IWebElement>(list);
        }

        public override string ToString()
        {
            return "BySelf";
        }
    }
}